maintainers/patches

```diff
@@ -0,0 +1,88 @@
+#!/bin/zsh
+
+PROGRAM=$0
+
+success() {
+  echo "\e[1;32mDone:\e[m $1"
+  exit 0
+}
+
+fail() {
+  echo "\e[1;31mError:\e[m $1"
+  exit 1
+}
+
+help() {
+  cat <<EOF
+Usage: $PROGRAM {apply|save}
+
+  apply       Applies the patches to the Tock submodule.
+              This resets Tock and can always be called.
+
+  save        Saves the Tock submodule to the patches.
+              Should only be called after apply and when all changes have been
+              added to a commit. After saving, you can run ./setup.sh to return
+              to normal state. Otherwise you can continue editing Tock and
+              calling save.
+EOF
+  exit 0
+}
+
+commit() {
+  local message=$1
+  git add .
+  git commit -qm$message
+}
+
+get_root() {
+  git ls-tree HEAD:third_party | sed -n 's/^.* \([^ ]\+\)\ttock$/\1/p'
+}
+
+get_head() {
+  git rev-parse HEAD
+}
+
+apply() {
+  git submodule -q update third_party/tock
+  ( cd third_party/tock
+    git reset -q --hard
+    git clean -qfxd
+    rsync -a ../../boards .
+    commit '00-boards'
+    for file in ../../patches/tock/*; do
+      patch -sp1 < $file
```

Comment:
Ok, I didn't observe any issue. It's maybe when patch don't apply very cleanly. I changed to `git apply` anyway because it should still work the same. The only difference we get are warnings about trailing spaces due to the `.jlink` files being DOS style (i.e. they use `\r\n` instead of `\n`). But I shut down the warnings which should be good enough.